qt/widgets/common/test/FitScriptGeneratorPresenterTest.h

```diff
@@ -0,0 +1,326 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidKernel/WarningSuppressions.h"
+#include "MantidQtWidgets/Common/FitScriptGeneratorPresenter.h"
+#include "MantidQtWidgets/Common/IFitScriptGeneratorModel.h"
+#include "MantidQtWidgets/Common/IFitScriptGeneratorView.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+
+#include <memory>
+
+using namespace MantidQt::MantidWidgets;
+using namespace testing;
+using namespace WorkspaceCreationHelper;
+
+GNU_DIAG_OFF_SUGGEST_OVERRIDE
+
+class MockFitScriptGeneratorView : public IFitScriptGeneratorView {
```

Comment:
is it worth putting the mock classes into a seperate file?

Reply:
These Mocks are only used in this test, so I would say it wouldn't be worth it.